File: address-validation-international-nodejs/REST/avi_response.js
/**
 * Input parameters for the GetAddressInfo API call.
 */
export class GetAddressInfoInput {
    constructor(data = {}) {
        this.Address1 = data.Address1;
        this.Address2 = data.Address2;
        this.Address3 = data.Address3;
        this.Address4 = data.Address4;
        this.Address5 = data.Address5;
        this.Locality = data.Locality;
        this.AdministrativeArea = data.AdministrativeArea;
        this.PostalCode = data.PostalCode;
        this.Country = data.Country;
        this.OutputLanguage = data.OutputLanguage || 'ENGLISH';
        this.LicenseKey = data.LicenseKey;
        this.IsLive = data.IsLive !== undefined ? data.IsLive : true;
        this.TimeoutSeconds = data.TimeoutSeconds !== undefined ? data.TimeoutSeconds : 15;
    }

    toString() {
        return `GetAddressInfoInput: Address1 = ${this.Address1}, Address2 = ${this.Address2}, Address3 = ${this.Address3}, Address4 = ${this.Address4}, Address5 = ${this.Address5}, Locality = ${this.Locality}, AdministrativeArea = ${this.AdministrativeArea}, PostalCode = ${this.PostalCode}, Country = ${this.Country}, OutputLanguage = ${this.OutputLanguage}, LicenseKey = ${this.LicenseKey}, IsLive = ${this.IsLive}, TimeoutSeconds = ${this.TimeoutSeconds}`;
    }
}

/**
 * Information Component for API responses.
 */
export class InformationComponent {
    constructor(data = {}) {
        this.Name = data.Name;
        this.Value = data.Value;
    }

    toString() {
        return `Name = ${this.Name}, Value = ${this.Value}`;
    }
}

/**
 * Error object for API responses.
 */
export class Error {
    constructor(data = {}) {
        this.Type = data.Type;
        this.TypeCode = data.TypeCode;
        this.Desc = data.Desc;
        this.DescCode = data.DescCode;
    }

    toString() {
        return `Error: Type = ${this.Type}, TypeCode = ${this.TypeCode}, Desc = ${this.Desc}, DescCode = ${this.DescCode}`;
    }
}

/**
 * Address information for a validated international address.
 */
export class AddressInfo {
    constructor(data = {}) {
        this.Status = data.Status;
        this.ResolutionLevel = data.ResolutionLevel;
        this.Address1 = data.Address1;
        this.Address2 = data.Address2;
        this.Address3 = data.Address3;
        this.Address4 = data.Address4;
        this.Address5 = data.Address5;
        this.Address6 = data.Address6;
        this.Address7 = data.Address7;
        this.Address8 = data.Address8;
        this.Locality = data.Locality;
        this.AdministrativeArea = data.AdministrativeArea;
        this.PostalCode = data.PostalCode;
        this.Country = data.Country;
        this.CountryISO2 = data.CountryISO2;
        this.CountryISO3 = data.CountryISO3;
        this.InformationComponents = (data.InformationComponents || []).map(component => new InformationComponent(component));
    }

    toString() {
        const componentsString = this.InformationComponents.length
            ? this.InformationComponents.map(component => component.toString()).join(', ')
            : 'null';
        return `AddressInfo: Status = ${this.Status}, ResolutionLevel = ${this.ResolutionLevel}, Address1 = ${this.Address1}, Address2 = ${this.Address2}, Address3 = ${this.Address3}, Address4 = ${this.Address4}, Address5 = ${this.Address5}, Address6 = ${this.Address6}, Address7 = ${this.Address7}, Address8 = ${this.Address8}, Locality = ${this.Locality}, AdministrativeArea = ${this.AdministrativeArea}, PostalCode = ${this.PostalCode}, Country = ${this.Country}, CountryISO2 = ${this.CountryISO2}, CountryISO3 = ${this.CountryISO3}, InformationComponents = [${componentsString}]`;
    }
}

/**
 * Response from GetAddressInfo API, containing validated address information.
 */
export class AddressInfoResponse {
    constructor(data = {}) {
        this.AddressInfo = data.AddressInfo ? new AddressInfo(data.AddressInfo) : null;
        this.Error = data.Error ? new Error(data.Error) : null;
    }

    toString() {
        return `AddressInfoResponse: AddressInfo = ${this.AddressInfo ? this.AddressInfo.toString() : 'null'}, Error = ${this.Error ? this.Error.toString() : 'null'}]`;
    }
}

export default AddressInfoResponse;

File: address-validation-international-python/REST/avi_response.py

from dataclasses import dataclass
from typing import Optional, List


@dataclass
class GetAddressInfoInput:
    Address1: Optional[str] = None
    Address2: Optional[str] = None
    Address3: Optional[str] = None
    Address4: Optional[str] = None
    Address5: Optional[str] = None
    Locality: Optional[str] = None
    AdministrativeArea: Optional[str] = None
    PostalCode: Optional[str] = None
    Country: Optional[str] = None
    OutputLanguage: str = "ENGLISH"
    LicenseKey: Optional[str] = None
    IsLive: bool = True
    TimeoutSeconds: int = 15

    def __str__(self) -> str:
        return (f"GetAddressInfoInput: Address1={self.Address1}, Address2={self.Address2}, Address3={self.Address3}, "
                f"Address4={self.Address4}, Address5={self.Address5}, Locality={self.Locality}, "
                f"AdministrativeArea={self.AdministrativeArea}, PostalCode={self.PostalCode}, "
                f"Country={self.Country}, OutputLanguage={self.OutputLanguage}, LicenseKey={self.LicenseKey}, "
                f"IsLive={self.IsLive}, TimeoutSeconds={self.TimeoutSeconds}")


@dataclass
class InformationComponent:
    Name: Optional[str] = None
    Value: Optional[str] = None

    def __str__(self) -> str:
        return f"InformationComponent: Name={self.Name}, Value={self.Value}"


@dataclass
class Error:
    Type: Optional[str] = None
    TypeCode: Optional[str] = None
    Desc: Optional[str] = None
    DescCode: Optional[str] = None

    def __str__(self) -> str:
        return f"Error: Type={self.Type}, TypeCode={self.TypeCode}, Desc={self.Desc}, DescCode={self.DescCode}"


@dataclass
class AddressInfo:
    Status: Optional[str] = None
    ResolutionLevel: Optional[str] = None
    Address1: Optional[str] = None
    Address2: Optional[str] = None
    Address3: Optional[str] = None
    Address4: Optional[str] = None
    Address5: Optional[str] = None
    Address6: Optional[str] = None
    Address7: Optional[str] = None
    Address8: Optional[str] = None
    Locality: Optional[str] = None
    AdministrativeArea: Optional[str] = None
    PostalCode: Optional[str] = None
    Country: Optional[str] = None
    CountryISO2: Optional[str] = None
    CountryISO3: Optional[str] = None
    InformationComponents: Optional[List['InformationComponent']] = None

    def __post_init__(self):
        if self.InformationComponents is None:
            self.InformationComponents = []

    def __str__(self) -> str:
        components_string = ', '.join(str(component) for component in self.InformationComponents) if self.InformationComponents else 'None'
        return (f"AddressInfo: Status={self.Status}, ResolutionLevel={self.ResolutionLevel}, "
                f"Address1={self.Address1}, Address2={self.Address2}, Address3={self.Address3}, "
                f"Address4={self.Address4}, Address5={self.Address5}, Address6={self.Address6}, "
                f"Address7={self.Address7}, Address8={self.Address8}, Locality={self.Locality}, "
                f"AdministrativeArea={self.AdministrativeArea}, PostalCode={self.PostalCode}, "
                f"Country={self.Country}, CountryISO2={self.CountryISO2}, CountryISO3={self.CountryISO3}, "
                f"InformationComponents=[{components_string}]")


@dataclass
class AddressInfoResponse:
    AddressInfo: Optional['AddressInfo'] = None
    Error: Optional['Error'] = None

    def __str__(self) -> str:
        address_info_string = str(self.AddressInfo) if self.AddressInfo else 'None'
        error_string = str(self.Error) if self.Error else 'None'
        return (f"AddressInfoResponse: AddressInfo={address_info_string}, "
                f"Error={error_string}]")